lib/rules/no-restricted-exports.js

```diff
@@ -27,27 +27,78 @@ module.exports = {
-                        type: "string"
+            anyOf: [
+                {
+                    type: "object",
+                    properties: {
+                        restrictedNamedExports: {
+                            type: "array",
+                            items: {
+                                type: "string"
+                            },
+                            uniqueItems: true
+                        }
                     },
-                    uniqueItems: true
+                    additionalProperties: false
+                },
+                {
+                    type: "object",
+                    properties: {
+                        restrictedNamedExports: {
+                            type: "array",
+                            items: {
+                                type: "string",
+                                pattern: "^(?!default$)"
+                            },
+                            uniqueItems: true
+                        },
+                        restrictDefaultExports: {
+                            type: "object",
+                            properties: {
+
+                                // Allow/Disallow `export default foo; export default 42; export default function foo() {}` format
+                                direct: {
+                                    type: "boolean"
+                                },
+
+                                // Allow/Disallow `export { foo as default };` declarations
+                                named: {
+                                    type: "boolean"
+                                },
+
+                                //  Allow/Disallow `export { default } from "mod"; export { default as default } from "mod";` declarations
+                                defaultFrom: {
+                                    type: "boolean"
+                                },
+
+                                //  Allow/Disallow `export { foo as default } from "mod";` declarations
+                                namedFrom: {
+                                    type: "boolean"
+                                },
+
+                                //  Allow/Disallow `export * as default from "mod"`; declarations
+                                namespaceFrom: {
+                                    type: "boolean"
+                                }
+                            },
+                            additionalProperties: false
+                        }
+                    },
+                    additionalProperties: false
                 }
-            },
-            additionalProperties: false
+            ]
         }],
 
         messages: {
-            restrictedNamed: "'{{name}}' is restricted from being used as an exported name."
+            restrictedNamed: "'{{name}}' is restricted from being used as an exported name.",
+            restrictedDefault: "Exporting the 'default' value is restricted."
```

Comment:
```suggestion
            restrictedDefault: "Exporting 'default' is restricted."
```

`default` is binding, like other exports. I think we can shorten the message to make it more technically correct.

Reply:
Updated 👍🏻